From d789158780c13f3a9b941f67159b5ead7dd84842 Mon Sep 17 00:00:00 2001
From: David Abdurachmanov <david.abdurachmanov@sifive.com>
Date: Tue, 28 Jan 2020 03:09:34 -0800
Subject: [PATCH 6/7] SiFive Unleashed GPIO restart

Add gpio-restart to allow rebooting machine from user land.

Signed-off-by: David Abdurachmanov <david.abdurachmanov@sifive.com>
Upstream-Status: Not posted for a review
---
 arch/riscv/boot/dts/sifive/fu540-c000.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/riscv/boot/dts/sifive/fu540-c000.dtsi b/arch/riscv/boot/dts/sifive/fu540-c000.dtsi
index 023a8fd..3453472 100644
--- a/arch/riscv/boot/dts/sifive/fu540-c000.dtsi
+++ b/arch/riscv/boot/dts/sifive/fu540-c000.dtsi
@@ -4,6 +4,7 @@
 /dts-v1/;
 
 #include <dt-bindings/clock/sifive-fu540-prci.h>
+#include <dt-bindings/gpio/gpio.h>
 
 / {
 	#address-cells = <2>;
@@ -287,5 +288,9 @@
 			clocks = <&prci PRCI_CLK_TLCLK>;
 			status = "disabled";
 		};
+		gpio-restart {
+			compatible = "gpio-restart";
+			gpios = <&gpio 10 GPIO_ACTIVE_LOW>;
+		};
 	};
 };
-- 
2.7.4

